### ISSUE TITLE
`tf.keras.callbacks.BackupAndRestore` showing wrong number of steps after restoring the model!

### ISSUE THREAD

**Issue by mohammadreza490** [tensorflow/tensorflow#55241]
<em>Please make sure that this is a bug. As per our
[GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md),
we only address code/doc bugs, performance issues, feature requests and
build/installation issues on GitHub. tag:bug_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): Yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): colab
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
- TensorFlow installed from (source or binary): colab
- TensorFlow version (use command below): v2.8.0-0-g3f878cff5b6 2.8.0
- Python version: 3.7.12
- Bazel version (if compiling from source):
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version: 11.2 
- GPU model and memory: Tesla V100-SXM2-16GB

You can collect some of this information using our environment capture
[script](https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh)
You can also obtain the TensorFlow version with:
1. TF 1.0: `python -c "import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"`
2. TF 2.0: `python -c "import tensorflow as tf; print(tf.version.GIT_VERSION, tf.version.VERSION)"`

**Describe the current behavior**
By using the BackupAndRestore, I can save and reload the model from the last saved checkpoint and the training will continue from the last epoch before the interruption. The problem I'm facing is that after the training of the epoch is complete, the number of steps for that epoch is wrong (it takes the number of steps of the epoch and multiplies it with the number of epochs completed). The time of the epoch is correct but because the number of steps is wrong, the time per step is also wrong (as It divides the time taken for this epoch and divides it with the wrong number of steps). This issue causes the next epochs to show wrong progress bar and wrong number of steps as well.

You can see an example here:
My model trains for 1481 steps per epoch and it usually takes 7 to 8 seconds per step therefore I expect the total time of the epoch to be around 1481 * 8 = 10000 to 12000 seconds. I get the following log for training the model after restoring the 16th epoch by using the last saved checkpoint.

Epoch 16/40
23696/23696 [==============================] - 11513s 484ms/step - loss: 32.1565 - lr: 0.0010
Epoch 17/40
   74/23696 [..............................] - ETA: 40:37:26 - loss: 31.8166

As you can see, the number of steps should be 1481 but it's instead 16*1481=23696 for 16th epoch. The time per step should be 11513 / 1481 which is around 7s but what it shows is 11513 / 23696 = 484ms.

Also, the next epoch has a wrong total number of steps and ETA as well because of the wrong number of steps of the previous epoch. it stays like this until I restore from another epoch which produces the same issue

**Describe the expected behavior**
The expected behavior is to see the total time of the epoch (and not all the trained epochs) and to see the correct time per step

For example, I expect to see this in my case:
Epoch 16/40
1481/1481 [==============================] - 11513s 7s/step - loss: 32.1565 - lr: 0.0010
Epoch 17/40
   74/1481 [..............................] - ETA: 3:19:00 - loss: 31.8166

**[Contributing](https://www.tensorflow.org/community/contribute)**

- Do you want to contribute a PR? (yes/no): No
- Briefly describe your candidate solution(if contributing):

**Standalone code to reproduce the issue**
Provide a reproducible test case that is the bare minimum necessary to generate
the problem. If possible, please share a link to Colab/Jupyter/any notebook.

This is the training method of my model.
```
def train(self, loss_function:Callable, data_generator:Callable=None, learning_rate_scheduler:tf.keras.callbacks=None, learning_rate:float = 1e-3, epochs:int=40, batch_size:int=8, shuffle_data:bool=True, seed:int=0)->History:
    '''This method will generate the train_dataset and trains and backup the model after each epoch.
       it finally saves the model and removes the checkpoints
    '''
    def store_spectrograms():
        music_names = os.listdir(self._train_dir_path)
        music_names.remove("spectrograms")
        train_spectrograms_temp_dir_path = os.path.join(self._train_dir_path, "spectrograms")
        for music_name in music_names:
            #print(f"storing spectrograms for {music_name}")
            music_dir_path = os.path.join(train_spectrograms_temp_dir_path, music_name)
            
            for file_name in ["mixture.wav", "vocals.wav", "accompaniment.wav"]:
                destination_dir = os.path.join(music_dir_path, file_name[:-4])
                if os.path.exists(destination_dir):
                    # print(f"{destination_dir} already exists")
                    continue
                Config_Handler.make_dir(destination_dir)
                wav_file_handler = Wav_File_Handler(wav_file_path = os.path.join(music_dir_path, file_name))
                _, spectrograms = wav_file_handler.generate_segments_and_spectrograms(segment_length_in_seconds=self._segment_length_in_seconds, clip_stft_to_even_frequency_bins=self._clip_stft_to_even_frequency_bins)
                for i, spectrogram in enumerate(spectrograms):
                    spectrogram_id = f"id_{i}"
                    destination_path = os.path.join(destination_dir, f"{spectrogram_id}-spectrogram.npy")
                    np.save(destination_path, spectrogram)
            #print(f"storing spectrograms for {music_name} completed!\n")

    store_spectrograms()

    if os.path.exists(self._model_path):
      train_more = str.lower(input(f"Model is already trained, are you sure you want to train it for {epochs} more epochs? \n (y for yes and n for no:)"))
      while train_more not in ["y", "n"]:
        print("invalid answer!")
        train_more = input(f"Model is already trained, are you sure you want to train it for {epochs} more epochs? \n (y for yes and n for no:)")
      if train_more == "n":
        print("cancelling training!")
        return
      else:
        print(f"model will be trained for {epochs} more epochs")

    input_file_name = "mixture.wav"
    output_vocal_file_name = "vocals.wav"
    output_ac_file_name = "accompaniment.wav"
    music_dict = dict()
    
    train_dataset = tf.data.Dataset.from_generator(lambda: self.generate_data(self._train_dir_path),
                                             output_signature=(tf.TensorSpec(shape=(None, self._input_shape[0], self._input_shape[1], self._input_shape[2]), dtype=tf.float32),
                                                               tf.TensorSpec(shape=(None, self._input_shape[0], self._input_shape[1], 2, self._input_shape[2]), dtype=tf.float32)))
    # test_dataset = tf.data.Dataset.from_generator(lambda: self.generate_data(self._test_dir_path, shuffle_data=shuffle_data, seed=seed),
    #                                          output_signature=(tf.TensorSpec(shape=(None, 1024, 256, 1), dtype=tf.float32),
    #                                                            tf.TensorSpec(shape=(None, 1024, 256, 2, 1), dtype=tf.float32)))
    optimizer = AdamW(weight_decay=1e-6, learning_rate=1e-3)

    tensorboard_callback = tf.keras.callbacks.TensorBoard(
    log_dir=self._tensorboard_logs_dir, histogram_freq=0, write_graph=True,
    write_images=False, write_steps_per_second=False, update_freq='epoch',
    profile_batch=0, embeddings_freq=0, embeddings_metadata=None,)
    #tf.debugging.experimental.enable_dump_debug_info(self._tensorboard_logs_dir, tensor_debug_mode="FULL_HEALTH", circular_buffer_size=-1)
    #tf.debugging.experimental.disable_dump_debug_info()

    backup_and_restore_callback = tf.keras.callbacks.BackupAndRestore( 
    backup_dir=self._checkpoint_path)
    callbacks = [tensorboard_callback, backup_and_restore_callback] + [tf.keras.callbacks.LearningRateScheduler(learning_rate_scheduler)] if learning_rate_scheduler != None else []
    self._model.compile(loss=loss_function, optimizer=optimizer)
    print("starting training!")
    history = self._model.fit(train_dataset, epochs=epochs, batch_size=batch_size, callbacks=callbacks, verbose=1)

    self.save_model()
    self._finalise_training()

    return history
```

**Other info / logs** Include any logs or source code that would be helpful to
diagnose the problem. If including tracebacks, please include the full
traceback. Large logs and files should be attached.

**Comment by sushreebarsa** [comment]
@mohammadreza490 
Please post this issue on [keras-team/keras repo.](https://github.com/keras-team/keras/issues)
To know more see;
[https://discuss.tensorflow.org/t/keras-project-moved-to-new-repository-in-https-github-com-keras-team-keras/1999](https://discuss.tensorflow.org/t/keras-project-moved-to-new-repository-in-https-github-com-keras-team-keras/1999)
Thanks!

**Comment by mohammadreza490** [comment]
Will do thank you!
